pico/src/temp.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2024 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/i2c.h>
+
+#define DEV_ADDR   (0x4e)
+#define START_ADDR (0x00)
+#define RESOLUTION (0.0625)
+#define BIT_SHIFT  (4)
+
+int get_temp(const struct device *dev, float *temp)
+{
+	int ret;
+	uint8_t tmp[2];
+
+	if (temp == NULL) {
+		return -EINVAL;
+	}
+
+	ret = i2c_burst_read(dev, DEV_ADDR, START_ADDR, tmp, sizeof(tmp));
+	if (ret == 0) {
+		tmp[1] = tmp[1] >> BIT_SHIFT;
+		*temp = (int8_t)tmp[0] + (float)tmp[1] * RESOLUTION;
```

Comment:
this line is chaos.  `float pointer` = `uint8_ casted to int8_t` + `uint8_t casted to float` * `double`
No need to fix it now this one was there and this commit just moved it.

Reply:
Yes, that's right.
We will fix it another PR.